api/v1beta1/hostedcluster_types.go

```diff
@@ -580,12 +580,20 @@ type CIDRBlock string
 
 // APIServerNetworking specifies how the APIServer is exposed inside a cluster
 // node.
+// +kubebuilder:validation:XValidation:rule="(has(self.advertiseAddress) && self.advertiseAddresses.size == 0) || (!has(self.advertiseAddress) && self.advertiseAddresses.size > 0)", message="AdvertiseAddress and AdvertiseAddresses are exclusive between themselves"
 type APIServerNetworking struct {
 	// AdvertiseAddress is the address that nodes will use to talk to the API
 	// server. This is an address associated with the loopback adapter of each
-	// node. If not specified, 172.20.0.1 is used.
+	// node. If not specified, 172.20.0.1 or fd02::1 are used.
 	AdvertiseAddress *string `json:"advertiseAddress,omitempty"`
 
+	// AdvertiseAddresses are the addresses that nodes will use to talk to the API
+	// server. These are one or more addresses associated with the loopback adapter of each
+	// node. If not specified, 172.20.0.1 or/and fd02::1 are used.
+	// This variable differs from the above one because this will be used in dual stak scenarios.
+	// NOTE: Both of them could be used at the same time and the controller will take care of them
```

Comment:
> // NOTE: Both of them could be used at the same time and the controller will take care of them
is this note still true with the CEL validation in place?

Reply:
Oh you're right, let me fix this